minitwit/src/main/java/WebApplication.java

```diff
@@ -387,6 +396,116 @@ public static void updateLatest(Request request) {
         return "";
     };
 
+    public static Route serveSimFllws = (Request request, Response response) -> {
+
+        //update_latest(request); //TODO: call
+
+        var db = new SQLite();
+        var conn = db.getConnection();
+
+        var who_id = getUserID(db, request.params(":username"));
+
+        if (who_id == 0) {
+            conn.close();
+            response.status(404);
+            return false;
+        }
+
+        Gson gson = new Gson();
+        Fllws fllws = gson.fromJson(request.body(), Fllws.class);
+
+
+        if (Objects.equals(request.requestMethod(), "POST") && !fllws.follow.isEmpty()) {
+
+            var whom_id = getUserID(db, fllws.follow);
+
+            if (whom_id == 0) {
+                conn.close();
+                response.status(404);
+                return false;
+            }
+
+
+            var insert = conn.prepareStatement("insert into follower (\n" +
+                    "                who_id, whom_id) values (?, ?)");
+
+            insert.setInt(1, who_id);
+            insert.setInt(2, whom_id);
+            insert.execute();
+
+            conn.close();
+
+            response.status(200);
+            return true;
+
+        }
+
+        if (Objects.equals(request.requestMethod(), "POST") && !fllws.unfollow.isEmpty()) {
+
+            var whom_id = getUserID(db, fllws.unfollow);
+
+            if (whom_id == 0) {
+                conn.close();
+                response.status(404);
+                return false;
+            }
+
+            var insert = conn.prepareStatement("delete from follower where who_id=? and whom_id=?");
+
+            insert.setInt(1, who_id);
+            insert.setInt(2, whom_id);
+            insert.execute();
+
+            conn.close();
+
+            response.status(200);
+            return true;
+        }
+
+        if (Objects.equals(request.requestMethod(), "GET")) {
+
+            var insert = conn.prepareStatement("SELECT user.username FROM user INNER JOIN follower ON follower.whom_id=? WHERE follower.who_id=? LIMIT ?");
+
+            insert.setInt(1, request.session().attribute("user_id"));
+            insert.setInt(2, who_id);
+            insert.setInt(3, 100);
```

Comment:
How come the SQL statement is different than the one in the reference simulator?
Also, the API uses basic auth and not sessions.

Reply:
I just deleted a long comment about how i couldnt spot the difference, but now i realize the mistake... i thought they had directly inserted the user_id, thought it looked a bit odd too, now i know shy. My mistake